inference-engine/tests/functional/plugin/cpu/single_layer_tests/reduce_ops.cpp

```diff
@@ -68,46 +118,77 @@ class ReduceCPULayerTest : public testing::WithParamInterface<ReduceLayerCPUTest
-            // We change the range of random values to avoid possible floating point overflow
-            auto blob = FuncTestUtils::createAndFillBlob(info.getTensorDesc(), 10, 5);
-            if (Precision::FP32 == info.getTensorDesc().getPrecision()) {
-                auto *rawBlobDataPtr = blob->buffer().as<float *>();
-                for (size_t i = 0; i < blob->size(); ++i) {
-                    rawBlobDataPtr[i] /= 10.f;
-                }
-            } else if (Precision::BF16 == info.getTensorDesc().getPrecision()) {
-                auto *rawBlobDataPtr = blob->buffer().as<ngraph::bfloat16 *>();
-                for (size_t i = 0; i < blob->size(); ++i) {
-                    rawBlobDataPtr[i] /= 10.f;
+
+    void generate_inputs(const std::vector<ngraph::Shape>& targetInputStaticShapes) override {
+        inputs.clear();
+        const auto& funcInputs = function->inputs();
+        for (int i = 0; i < funcInputs.size(); ++i) {
+            const auto& funcInput = funcInputs[i];
+            ov::runtime::Tensor tensor;
+            if (reductionType == ngraph::helpers::ReductionType::Prod) {
+                tensor = ov::test::utils::create_and_fill_tensor(funcInput.get_element_type(), targetInputStaticShapes[i], 10, 5);
+                if (netPrecision == ElementType::f32) {
+                    auto *rawBlobDataPtr = static_cast<float *>(tensor.data());
+                    for (size_t i = 0; i < tensor.get_size(); ++i) {
+                        rawBlobDataPtr[i] /= 10.f;
+                    }
+                } else if (netPrecision == ElementType::bf16) {
+                    auto *rawBlobDataPtr = static_cast<ngraph::bfloat16 *>(tensor.data());
+                    for (size_t i = 0; i < tensor.get_size(); ++i) {
+                        rawBlobDataPtr[i] /= 10.f;
+                    }
                 }
+            } else {
+                tensor = ov::test::utils::create_and_fill_tensor(funcInput.get_element_type(), targetInputStaticShapes[i]);
             }
-            return blob;
-        } else {
-            return LayerTestsCommon::GenerateInput(info);
+
+            inputs.insert({funcInput.get_node_shared_ptr(), tensor});
         }
     }
 
 private:
     ngraph::helpers::ReductionType reductionType;
+    ElementType netPrecision;
 };
 
 TEST_P(ReduceCPULayerTest, CompareWithRefs) {
     SKIP_IF_CURRENT_TEST_IS_DISABLED()
 
-    Run();
-    CheckPluginRelatedResults(executableNetwork, "Reduce");
+    run();
+
+    // TODO: Should be uncommented after updating the CheckPluginRelatedResults() method
+    // CheckPluginRelatedResults(executableNetwork, "Reduce");
 }
 namespace {
-std::vector<Precision> inpOutPrc = {Precision::BF16, Precision::FP32};
+// TODO: Should be uncommented after fixing "Interpreter backend doesn't implement evaluate method for OP Reduce*" in int_executable.cpp
+const std::vector<ElementType> inpOutPrc = {/*ElementType::bf16, */ElementType::f32};
```

Comment:
Tests already support bf16 validation. See Convolution test as an example.

Reply:
Uncommented. Tickets are added concerning tests code about Fusing per channel and bf16 compare threshold. No issues about functional code in CPU plugin Reduce node. Thanks Dmitry!